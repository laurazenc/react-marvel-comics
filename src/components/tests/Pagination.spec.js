import React from 'react';
import {shallow} from 'enzyme';
import {Pagination} from './../';


describe('<Pagination />', () => {

  function setPage() { }

  it('should render ', () => {
    const wrapper = shallow(<Pagination setPage={setPage} />);
    expect(wrapper).toMatchSnapshot();

  });
});
