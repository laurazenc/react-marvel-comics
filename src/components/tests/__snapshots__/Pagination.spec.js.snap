exports[`<Pagination /> should render  1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <ul
    className="pagination">
    <li
        className="disabled">
        <a
            onClick={[Function]}>
            «
        </a>
    </li>
    <li
        className="disabled">
        <a
            onClick={[Function]}>
            ⟨
        </a>
    </li>
    <li
        className="active">
        <a
            onClick={[Function]}>
            1
        </a>
    </li>
    <li
        className="">
        <a
            onClick={[Function]}>
            2
        </a>
    </li>
    <li
        className="">
        <a
            onClick={[Function]}>
            3
        </a>
    </li>
    <li
        className="">
        <a
            onClick={[Function]}>
            4
        </a>
    </li>
    <li
        className="">
        <a
            onClick={[Function]}>
            5
        </a>
    </li>
    <li
        className="">
        <a
            onClick={[Function]}>
            6
        </a>
    </li>
    <li
        className="">
        <a
            onClick={[Function]}>
            7
        </a>
    </li>
    <li
        className="">
        <a
            onClick={[Function]}>
            8
        </a>
    </li>
    <li
        className="">
        <a
            onClick={[Function]}>
            9
        </a>
    </li>
    <li
        className="">
        <a
            onClick={[Function]}>
            10
        </a>
    </li>
    <li
        className="">
        <a
            onClick={[Function]}>
            ⟩
        </a>
    </li>
    <li
        className="">
        <a
            onClick={[Function]}>
            »
        </a>
    </li>
</ul>,
  "nodes": Array [
    <ul
      className="pagination">
      <li
            className="disabled">
            <a
                  onClick={[Function]}>
                  «
            </a>
      </li>
      <li
            className="disabled">
            <a
                  onClick={[Function]}>
                  ⟨
            </a>
      </li>
      <li
            className="active">
            <a
                  onClick={[Function]}>
                  1
            </a>
      </li>
      <li
            className="">
            <a
                  onClick={[Function]}>
                  2
            </a>
      </li>
      <li
            className="">
            <a
                  onClick={[Function]}>
                  3
            </a>
      </li>
      <li
            className="">
            <a
                  onClick={[Function]}>
                  4
            </a>
      </li>
      <li
            className="">
            <a
                  onClick={[Function]}>
                  5
            </a>
      </li>
      <li
            className="">
            <a
                  onClick={[Function]}>
                  6
            </a>
      </li>
      <li
            className="">
            <a
                  onClick={[Function]}>
                  7
            </a>
      </li>
      <li
            className="">
            <a
                  onClick={[Function]}>
                  8
            </a>
      </li>
      <li
            className="">
            <a
                  onClick={[Function]}>
                  9
            </a>
      </li>
      <li
            className="">
            <a
                  onClick={[Function]}>
                  10
            </a>
      </li>
      <li
            className="">
            <a
                  onClick={[Function]}>
                  ⟩
            </a>
      </li>
      <li
            className="">
            <a
                  onClick={[Function]}>
                  »
            </a>
      </li>
</ul>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Pagination
        setPage={[Function]} />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Pagination {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "setPage": [Function],
        },
        "refs": Object {},
        "state": Object {
          "pager": Object {
            "currentPage": 1,
            "endIndex": 10,
            "endPage": 10,
            "pageSize": 10,
            "pages": Array [
              1,
              2,
              3,
              4,
              5,
              6,
              7,
              8,
              9,
              10,
            ],
            "startIndex": 0,
            "startPage": 1,
            "totalPages": 74,
          },
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <ul
          className="pagination">
          <li
                    className="disabled">
                    <a
                              onClick={[Function]}>
                              «
                    </a>
          </li>
          <li
                    className="disabled">
                    <a
                              onClick={[Function]}>
                              ⟨
                    </a>
          </li>
          <li
                    className="active">
                    <a
                              onClick={[Function]}>
                              1
                    </a>
          </li>
          <li
                    className="">
                    <a
                              onClick={[Function]}>
                              2
                    </a>
          </li>
          <li
                    className="">
                    <a
                              onClick={[Function]}>
                              3
                    </a>
          </li>
          <li
                    className="">
                    <a
                              onClick={[Function]}>
                              4
                    </a>
          </li>
          <li
                    className="">
                    <a
                              onClick={[Function]}>
                              5
                    </a>
          </li>
          <li
                    className="">
                    <a
                              onClick={[Function]}>
                              6
                    </a>
          </li>
          <li
                    className="">
                    <a
                              onClick={[Function]}>
                              7
                    </a>
          </li>
          <li
                    className="">
                    <a
                              onClick={[Function]}>
                              8
                    </a>
          </li>
          <li
                    className="">
                    <a
                              onClick={[Function]}>
                              9
                    </a>
          </li>
          <li
                    className="">
                    <a
                              onClick={[Function]}>
                              10
                    </a>
          </li>
          <li
                    className="">
                    <a
                              onClick={[Function]}>
                              ⟩
                    </a>
          </li>
          <li
                    className="">
                    <a
                              onClick={[Function]}>
                              »
                    </a>
          </li>
</ul>,
        "_debugID": 2,
        "_renderedOutput": <ul
          className="pagination">
          <li
                    className="disabled">
                    <a
                              onClick={[Function]}>
                              «
                    </a>
          </li>
          <li
                    className="disabled">
                    <a
                              onClick={[Function]}>
                              ⟨
                    </a>
          </li>
          <li
                    className="active">
                    <a
                              onClick={[Function]}>
                              1
                    </a>
          </li>
          <li
                    className="">
                    <a
                              onClick={[Function]}>
                              2
                    </a>
          </li>
          <li
                    className="">
                    <a
                              onClick={[Function]}>
                              3
                    </a>
          </li>
          <li
                    className="">
                    <a
                              onClick={[Function]}>
                              4
                    </a>
          </li>
          <li
                    className="">
                    <a
                              onClick={[Function]}>
                              5
                    </a>
          </li>
          <li
                    className="">
                    <a
                              onClick={[Function]}>
                              6
                    </a>
          </li>
          <li
                    className="">
                    <a
                              onClick={[Function]}>
                              7
                    </a>
          </li>
          <li
                    className="">
                    <a
                              onClick={[Function]}>
                              8
                    </a>
          </li>
          <li
                    className="">
                    <a
                              onClick={[Function]}>
                              9
                    </a>
          </li>
          <li
                    className="">
                    <a
                              onClick={[Function]}>
                              10
                    </a>
          </li>
          <li
                    className="">
                    <a
                              onClick={[Function]}>
                              ⟩
                    </a>
          </li>
          <li
                    className="">
                    <a
                              onClick={[Function]}>
                              »
                    </a>
          </li>
</ul>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Pagination
    setPage={[Function]} />,
}
`;
